lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -177,3 +188,83 @@ def publish_event(event_name, result, **kwargs):
+        """
+        Collects any information that's necessary to execute this
+        transformer's transform method.
+        """
+        block_structure.request_xblock_fields('mode')
+        block_structure.request_xblock_fields('max_count')
+
+        ContentLibraryTransformer.set_block_analytics_summary(block_structure, modulestore())
+
+    def transform(self, usage_info, block_structure):
+        """
+        Transforms the order of the children of the randomized content block
+        to match the order of the selections made and stored in the XBlock 'selected' field.
+        """
+        for block_key in block_structure:
+            if block_key.block_type != 'library_content':
+                continue
+
+            library_children = block_structure.get_children(block_key)
+
+            if library_children:
+                state_dict = get_student_module_as_dict(usage_info.user, usage_info.course_key, block_key)
+                current_children_blocks = set(block.block_id for block in library_children)
+                current_selected_blocks = set(item[1] for item in state_dict['selected'])
+
+                # Ensure that the current children blocks are the same as the stored selections
+                # before ordering the blocks.
+                if current_children_blocks != current_selected_blocks:
+                    logger.info(
+                        u'Mismatch between the children of %s in the stored state and the actual children for user %s',
+                        str(block_key),
+                        usage_info.user.username
+                    )
+                    previous_count = len(current_selected_blocks.intersection(current_children_blocks))
+                    mode = block_structure.get_xblock_field(block_key, 'mode')
+                    max_count = block_structure.get_xblock_field(block_key, 'max_count')
+                    block_keys = LibraryContentModule.make_selection(
+                        state_dict['selected'], library_children, max_count, mode
+                    )
+                    state_dict['selected'] = block_keys['selected']
+                    random.shuffle(state_dict['selected'])
+                    StudentModule.save_state(
+                        student=usage_info.user,
+                        course_id=usage_info.course_key,
+                        module_state_key=block_key,
+                        defaults={
+                            'state': json.dumps(state_dict)
+                        },
+                    )
+                    ContentLibraryTransformer.publish_events(
+                        block_structure, block_key, previous_count, max_count, block_keys, usage_info.user.id
+                    )
+
+                ordering_data = {block[1]: position for position, block in enumerate(state_dict['selected'])}
+                library_children.sort(key=lambda block, data=ordering_data: data[block.block_id])
```

Comment:
@lgp171188, @bradenmacdonald: This failed with: `builtins:KeyError: 'ecdf45ce666efed87d8b'`

It was a request to the `/api/courses/v2/blocks/` endpoint. Digging for more specifics on the data...